v2/devices/violet.yml

```diff
@@ -0,0 +1,142 @@
+name: "Xiaomi Redmi Note 7 Pro"
+codename: "violet"
+formfactor: "phone"
+aliases: []
+doppelgangers: []
+user_actions:
+  recovery:
+    title: "Reboot to Recovery"
+    description: "With the device powered off, hold Volume Up + Power."
+    image: "phone_power_up"
+    button: true
+  bootloader:
+    title: "Reboot to Bootloader"
+    description: "With the device powered off, hold Volume Down + Power."
+    image: "phone_power_down"
+    button: true
+  boot:
+    title: "Boot the device"
+    description: "Power on the device."
+    image: "phone_power_up"
+    button: true
+  confirm_model:
+    title: "Confirm your model"
+    description: "Please double-check that your device is a Xiaomi Redmi Note 7 Pro (violet)."
+  unlock:
+    title: "OEM unlock"
+    description: "If you haven't done so already, make sure to OEM unlock your device first."
+    link: "https://en.miui.com/unlock/"
+unlock:
+  - "confirm_model"
+  - "unlock"
+handlers:
+  bootloader_locked:
+    actions:
+      - fastboot:oem_unlock:
+operating_systems:
+  - name: "Ubuntu Touch"
+    options:
+      - var: "channel"
+        name: "Channel"
+        tooltip: "The release channel"
+        link: "https://docs.ubports.com/en/latest/about/process/release-schedule.html"
+        type: "select"
+        remote_values:
+          systemimage:channels:
+      - var: "wipe"
+        name: "Wipe Userdata"
+        tooltip: "Wipe personal data (required if the previously installed OS is Android, MIUI or LineageOS)"
+        type: "checkbox"
+      - var: "bootstrap"
+        name: "Bootstrap"
+        tooltip: "Flash system partitions using fastboot"
+        type: "checkbox"
+        value: true
+    prerequisites: []
+    steps:
+      - actions:
+          - core:manual_download:
+              group: "firmware"
+              file:
+                name: "vendor.img"
+                url: "https://www.androidfilehost.com/?fid=2188818919693784460"
```

Comment:
Just an idea, can we rehost this file on some GitHub repo (maybe outside of UBPorts org) download section so it can be downloaded automatically by installer? Having manual download for vendor.img which people need to extract manually is a common source of headache when helping with installation on lavender.

Reply:
Is the installer able to handle zip files and unzip them automatically? If not, I'm not sure this would be a good change, as the .img size is 2GiB, if i remember correctly.

Reply:
@mardy yes, but will need extra step, see https://github.com/ubports/installer-configs/blob/master/v2/devices/yggdrasil.yml#L261

Reply:
Oh, nice! Then I can definitely do that :-)

BTW, since you are here :-), I'll ask you something else: should I also ship and flash the following files?

```
bluetooth.emmc.win
boot.emmc.win
dsp.emmc.win
efs1.emmc.win
efs2.emmc.win
efsc.emmc.win
efsg.emmc.win
modem.emmc.win
```
(I don't even remember how I got them, whether they are from a TWRP backup or something else...)

Reply:
Those look like they are from a TWRP backup from names. If you're feeling adventurous, you can check if what is needed when downgrading from stock Android 10. https://xiaomifirmwareupdater.com/ this unofficial website is pretty helpful for finding stock firmware fastboot packages.